File: savingforretirement.cpp
#include <iostream>
#include <math.h>
using namespace std;

int main() {
	int b, br, bs, a, as;
	cin >> b >> br >> bs >> a >> as;
	int min_saving = (br - b) * bs;
	int years = ceil((min_saving + 0.0) / as + a);
	if (((years - a) * as) <= min_saving)
		cout << (years + 1);
	else cout << years;
	return 0;
}

File: savingforretirement.hs
main :: IO()
main = do
 line <- getLine
 let [b, br, bs, a, as] = map read $ words $ line
 let minSaving = bs * (br - b)
 let (years, remain) = minSaving `divMod` as
 let ans = years + a + (if remain == 0 then 0 else 1)
 if (ans - a) * as <= minSaving
 then print (ans + 1)
 else print ans
